src/include/rados.h

```diff
@@ -517,6 +517,7 @@ struct ceph_osd_op {
 			__le64 ver;     /* no longer used */
 			__u8 op;	/* CEPH_OSD_WATCH_OP_* */
 			__u32 gen;      /* registration generation */
+			__u32 timeout; /* connection timeout */
```

Comment:
i was wondering how we handled the backward/forward compatibility here.  the op struct is wrapped in `MOSDOp::ops::op`. and `ceph_osd_op` is encoded using the raw encoder, i.e. just copy the memory chunk. so the maximum size of the member of union in `ceph_osd_op` is `sizeof(writesame)` = (64 + 64 + 64) bits. in other words, as long as the size of all struct are less or equal to 24 bytes, we are safe.

could you add a 
```c++
static_assert(sizeof(ceph_osd_op) == (2+4+(8+8+8)+4),
              "sizeof(ceph_osd_op) breaks the compatibility");
```
somewhere in this header file in a separated commit? in which,
```
 (2+4+(8+8+8)+4) =    (sizeof(ceph_osd_op::op) +
                                      sizeof(ceph_osd_op::flags) +
                                      sizeof(ceph_osd_op::writesame) +
                                      sizeof(ceph_osd_op::payload_len))
```

@athanatos am i right?